compiler/src/dotty/tools/dotc/reporting/UniqueMessagePositions.scala

```diff
@@ -23,7 +23,7 @@ trait UniqueMessagePositions extends Reporter {
         for (pos <- m.pos.start to m.pos.end) {
           positions get (ctx.source, pos) match {
             case Some(level) if level >= m.level => shouldHide = true
-            case _ => positions((ctx.source, pos)) = m.level
+            case _ => // positions((ctx.source, pos)) = m.level
```

Comment:
The CI fails because the change of this line. There's some bug hidden in `isHidden`, I have to comment it out to temporarily make sure all the exhaustivity/unreachability warnings are shown. Otherwise, the unreachable warning in the PR example will be incorrectly hidden.